pci/src/root.rs

```diff
@@ -250,29 +251,37 @@ impl PciConfigMmio {
     }
 }
 
-impl BusDevice for PciConfigMmio {
-    fn read(&mut self, offset: u64, data: &mut [u8]) {
+impl Device for PciConfigMmio {
+    fn name(&self) -> String {
+        "PciConfigMmio".to_string().clone()
+    }
+
+    fn read(&mut self, offset: GuestAddress, data: &mut [u8], _io_type: IoType) {
         // Only allow reads to the register boundary.
-        let start = offset as usize % 4;
+        let start = offset.0 as usize % 4;
         let end = start + data.len();
-        if end > 4 || offset > u64::from(u32::max_value()) {
+        if end > 4 || offset.0 > u64::from(u32::max_value()) {
```

Comment:
`raw_value` anywhere you're calling `.0` on a `GuestAddress`

Reply:
Got it! Will change that.